pygmt/xarray/backend.py

```diff
@@ -0,0 +1,116 @@
+"""
+An xarray backend for reading raster grid/image files using the 'gmt' engine.
+"""
+
+from typing import Literal
+
+import xarray as xr
+from pygmt._typing import PathLike
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import build_arg_list
+from pygmt.src.which import which
+from xarray.backends import BackendEntrypoint
+
+
+class GMTBackendEntrypoint(BackendEntrypoint):
+    """
+    Xarray backend to read raster grid/image files using 'gmt' engine.
+
+    Internally, GMT uses the netCDF C library to read netCDF files, and GDAL for GeoTIFF
+    and other raster formats. See also
+    :gmt-docs:`reference/features.html#grid-file-format`.
```

Comment:
Maybe add a few sentences explaining why users may want to read netCDF/GeoTIFF files via GMT, rather than netcdf4/rasterio, something like:

Compared to the "netcdf4"/"rasterio" engines, the "gmt" engine can read GMT remote files (file names starting with `@`) directly and provides the GMTDataArray accessor `.gmt` for easy access to GMT-specific features and metadata.

Reply:
Ok, updated the description in commit acab8b2cfbb2260960fccef6201dcc215980cad9, please review. I didn't mention the `netcdf4`/`rasterio` engines directly, but just mentioned that the GMT engine works better with GMT remote files and features in general.